src/openrct2/actions/WallPlaceAction.hpp

```diff
@@ -364,14 +391,14 @@ DEFINE_GAME_ACTION(WallPlaceAction, GAME_COMMAND_PLACE_WALL, GameActionResult)
         {
             if (!WallCheckObstruction(wallEntry, targetHeight / 8, clearanceHeight, &wallAcrossTrack))
             {
-                return MakeResult(
-                    GA_ERROR::NO_CLEARANCE, STR_CANT_BUILD_PARK_ENTRANCE_HERE, gGameCommandErrorText, gCommonFormatArgs);
+                return std::make_unique<WallPlaceActionResult>(
+                    GA_ERROR::NO_CLEARANCE, gGameCommandErrorText, gCommonFormatArgs);
             }
         }
 
         if (!map_check_free_elements_and_reorganise(1))
         {
-            return MakeResult(GA_ERROR::NO_FREE_ELEMENTS, STR_CANT_BUILD_PARK_ENTRANCE_HERE, gGameCommandErrorText);
+            return std::make_unique<WallPlaceActionResult>(GA_ERROR::NO_FREE_ELEMENTS, gGameCommandErrorText);
```

Comment:
MakeResult already returns the correct version, why was this changed? MakeResult should be used in all those cases.

Reply:
I was following along to the process of #8726 which was similar in this issue, MakeResult needs to be updated to call the constructor of my Subclass in order to fix the issue in #9054 , unless I am mistaken

Reply:
It should already use your subclass if you specified it via DEFINE_GAME_ACTION, it will use the template parameter.

Reply:
@duncanspumpkin Should I revert back to using MakeResult?

Reply:
I wasn't overly fussed which way it was done since there isn't consistency across the other game actions right now. It can either be reverted to make result or we can catch all the game actions in a make result pr

Reply:
Works for me, however I just pointed it out as those changes are somewhat unnecessary given it already has MakeResult, but fine.